File: Agencia-Conta-main/src/main/java/br/com/Agencia/Conta/entity/Transacao.java
package br.com.Agencia.Conta.entity;

import br.com.Agencia.Conta.entity.enums.TipoEnum;
import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Table(name = "tb_transacao")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Transacao {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @JsonFormat(pattern = "dd/MM/yyyy")
    private LocalDate dataOperacao;
    @Enumerated(EnumType.STRING)
    private TipoEnum tipo; // "Credito" ou "Debito"
    private Double valor;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    @PrePersist
    public void prePersist() {
        setDataOperacao(LocalDate.now());
    }
}

File: Agencia-Conta-app/src/app/transacao/transacao.ts
export class Transacao{
    id: number;
    dataOperacao: string;
    tipo: string;
    valor: number;
    idcliente: number;

    constructor(id: number = 0, dataOperacao: string, tipo: string = '', valor: number = 0, idcliente: number) {
        this.id = id;
        this.dataOperacao = dataOperacao;
        this.tipo = tipo;
        this.valor = valor;
        this.idcliente = idcliente;
    }
}